doc/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,422 @@
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| IP                     |  Internet Protocol       |
+| VRF                     | Virtual routing and forwarding
+
+
+# 1 Feature Overview
+
+The documents covers the various interfaces for configuring static routes using SONiC management framework.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide ability to configure IPv4 and IPv6 static routes using SONiC management framework.
+
+### 1.1.2 Configuration and Management Requirements
+ - Implement Static ip route CLI Commands.
+ - REST set/get support for Static IPv4 and IPv6 routes.
+ - gNMI set/get support for Static IPv4 and IPv6 routes.
+
+### 1.1.3 Scalability Requirements
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+### 1.2.2 Container
+### 1.2.3 SAI Overview
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+## 2.2 Functional Description
+# 3 Design
+## 3.1 Overview
+An existing table STATIC_ROUTE (which is not used currently) will be used to write static route from the transformer for any CLI, rest or gNMI request. This table will be monitored by bgpcfgd daemon and the config will be sent to vtysh shell for configuring in FRR.
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+STATIC_ROUTE table in config DB will be used to  support this feature.
+
+#### 3.2.1.1 STATIC_ROUTE
+```JSON
+;Defines IP static route  table
+;
+;Status: stable
+
+key                 = STATIC_ROUTE|vrf-name|prefix ;
+vrf-name            = 1\*15VCHAR ; VRF name
+prefix              = IPv4Prefix / IPv6prefix
+nexthop             = IPv4Address / IPv6Address; List of gateway addresses;
+ifname              = List of interfaces
+distance            = {0..255};List of distances.
+                      Its a Metric used to specify preference of next-hop
+                      if this distance is not set, 0 will be set to maintain the set;
+nexthop-vrf         =  1\*15VCHAR ; list of next-hop VRFs. It should be set only if ifname or nexthop IP  is not
+                       in the current VRF . The value is set to VRF name
+                       to which the interface or nexthop IP  belongs for route leaks.
+blackhole           =  List of boolean; true if the next-hop route is blackholed.                     
+```
+The nexthop-vrf if set, will allow to create a leaked route in the current VRF(vrf-name). The vrf-name, nexthop-vrf, prefix, ifname, distance and blackhole are all parameters required to configure static routes and route leaks in vtysh.
+If an interface is moved from one VRF to another and it exist in the STATIC_ROUTE table, error will be returned by the backend. User is expected to remove all static routes pertaining to the interface before binding the interface to a different VRF.
+If prefix is IPv4Prefix nexthop will have  IPv4Address and if  prefix is IPv6Prefix nexthop will have  IPv6Address.

```

Comment:
May be rewrite the wording to, "prefix and nexthop is expected to have the same address family"

Reply:
sure